### ISSUE TITLE
Improve Blog page

### ISSUE THREAD

**Issue by popovayoana** [ckan/ckan.org#42]
**Description**
Corporate blogs are "de rigueur" in the social and content marketing worlds as vehicles to build authority, share insights, foster community and provide value. We’ve already got the basics down, but we can take it to the next level.

**What's not working on our Blog page in my opinion**
- Navigation
  - **Problem:** No clear navigation (for readers visiting our blog, clear navigation can keep them browsing the site instead of bouncing if they don’t see the information they need right away... currently, the structure of the blog page doesn't allow for quick navigation) 
  - **Solution:** Implement categories (one easy way to add clarity would be e.g. adding “Categories” widget on the sidebar)
  - **Solution 2:** Insert a search bar _(updated)_
- Sharing
  - **Problem:** No sharing option (currently readers cannot share directly blog posts on social media)
  - **Solution:** Add "Share this" option (Tweet this, share via Facebook, LinkedIn)
- Readability
  - **Problem:** Some of the articles are too long and might be overwhelming
  - **Solution:** Use pull quotes (wherever possible of course) to break up long blog posts to make them more visually appealing
- General look (this is not quick fix but major one)
  - **Problem:** Design looks jejune, not juicy
  - **Solution:** Redesign so that it looks juicier (references: https://blog.google, https://www.thinkwithgoogle.com)

## Tasks
- [ ] decide what's possible in terms of capacity - quick fixes mentioned above or major fixes (like general look)
- [ ] categories
  - [ ] create a list of categories we want on the blog (note they should be different than tags)
  - [ ] come up with a navigation solution - how are we going to implement the categories? is it gonna be just a side widget or bigger redesign?
- [ ] implement "share this" option
- [ ] come up with a design for pull quotes (mockup?)
- [ ] ...

**Comment by popovayoana** [comment]
@Alex-Pavlyuk this is something that needs a deeper discussion, i dropped some initial thoughts...

**Comment by Alex-Pavlyuk** [comment]
@popovayoana I'm waiting for fixes for other tasks then I'll now better about our capacity

**Comment by popovayoana** [comment]
@Alex-Pavlyuk ok, let me know once you know more

**Comment by popovayoana** [comment]
@Alex-Pavlyuk what's the status of that issue? are we going to have a bombastic blog soon? : )))

**Comment by Alex-Pavlyuk** [comment]
@popovayoana I try to organize more active work on the site tasks this month, we allocate a developer who will be focused only on the site work but he has to examine the code before his work will be really conspicuous

**Comment by popovayoana** [comment]
@Alex-Pavlyuk understood, sounds good!

**Comment by alexmorev** [comment]
Hi, @popovayoana !
Please test the last changes in Blog pages.
The applied changes:
1. New model “Category” was created in database
2. Added connection of this model with BlogPostPage model (via Foreignkey)
3. Added group of menu items “Blog” with subitems “Posts”, “Categories” to admin page
4. Created posts list page with filters (Need to setup appropriate permissions to display the whole list of posts or the list for owner. I've already done it.)
5. Created navigation bar for Blog page with social share links, add new post button and category filter in user interface.
6. Created a new page for posts lists filtered by a chosen category
7. Added social share buttons to Post page

**Comment by popovayoana** [comment]
@Alex-Pavlyuk Great, looks good! Can we do two design changes:
1. remove the lines above and below the categories-filter box and social media icons? 
2. can we make the shape of the categories-filter box the same as the one we've used to circle "View on Github"? (for consistency purposes)

![image](https://user-images.githubusercontent.com/78435662/193113761-14f93b31-85c1-46bf-8388-2db26594bc64.png)

**Comment by Alex-Pavlyuk** [comment]
@popovayoana its done

**Comment by popovayoana** [comment]
@Alex-Pavlyuk @alexmorev looks better now, thanks! what's left IMO:

- navigation
  - categories **fixed** (an even better solution would be to have a beautiful widget with categories on aside but I guess that's not a priority atm)
  - search bar **not fixed**
- sharing
  - share this **fixed**
- readability
  - add pull quotes **not fixed**
- general look
  - redesign so that it looks juicier **not fixed** (that's not a quick fix so not a priority atm but definitely something we'd want to do in the future)

**Comment by alexmorev** [comment]
@popovayoana I've added a search bar to the blog page and hope it will be deployed on prod asap. It searches in titles, subtitles and post bodies. So after testing please let me know about any necessary improvements.

**Comment by popovayoana** [comment]
@Alex-Pavlyuk @alexmorev happy to close the issue but noticed one small improvement for the search functionality. I had some trouble locating the search functionality on the blog at first. It was only after a bit of exploration that I discovered the ckan-logo-like button on the left of the screen.

suggestion for improvement:

- **design & behavior**: the search button should mirror the social media icons on the right - styled as a half eclipse, same behaviour. the only change would be that instead of  social media icon we should have a search icon (a magnifying glass)
- **position:** the location of the search button now is quite effective - let's keep it on the left side, but align it exactly with the 'Share on Facebook' button on the right for symmetry and balance
- **interaction:** on hover, the eclipse should unfold, similar to the social media buttons' behavior
- **functionality:** on click, the current left menu should pop up as it does now - no changes here

![image](https://github.com/ckan/ckan.org/assets/78435662/f5f80751-44ef-47e5-a4c9-34fc24f36f57)

![image](https://github.com/ckan/ckan.org/assets/78435662/a865d4a3-0228-4af1-afca-9a7e651bc28c)

**Comment by alexmorev** [comment]
@popovayoana It's done. Please check. Waiting for your feedback.

**Comment by popovayoana** [comment]
@alexmorev i had to clean up my cache to see the end result properly 😅 it looks great, well done, and thanks! ❤️ 

before cleaning up my cache
![image](https://github.com/ckan/ckan.org/assets/78435662/9303a72b-eff2-44e3-a521-dda9669f7ad7)

after cleaning it up
![image](https://github.com/ckan/ckan.org/assets/78435662/575526ea-a172-4da7-8d6d-2c5248e5d50f)

**Comment by popovayoana** [comment]
FIXED by @alexmorev.
